### DISCUSSION TITLE
Is the nn~ external compatible with Max RNBO?

### DISCUSSION THREAD

**Discussion by stooby** [acids-ircam/RAVE#244] [General]
Has anyone tried integrating the `nn~` external into a Max [RNBO](https://rnbo.cycling74.com/) patch and exporting it to one of the supported targets (e.g. Javascript for a Web app, C++, VST3, etc)?

But since there's no mention of RNBO support that I could find in the RAVE documentation, I assume maybe the `nn~` external is not RNBO compatible at this time?

If it's not compatible with RNBO, then consider this a +1 for making it so.

Obviously it would be really cool if RAVE models could be exported to all the additional platforms RNBO export supports. Thanks.

**Comment by wwerkk** [comment]
I believe there's Scyclone which is a VST made with RNBO/JUCE, might be worth to check out the source code:
https://github.com/Torsion-Audio/Scyclone

In some part it seems to have been developed with RNBO/Max, but running the model seems to be done further down the line, if I am reading the code correctly (see: [onnxruntime](https://github.com/olilarkin/ort-builder)).

btw. I might have missed something, but are there any examples of RNBO compatible external Max objects? I know you can export RNBO as Max externals, but no clue about compatibility the other way round.

**Reply by stooby** [reply]
Thanks for the quick reply. I'll check out Scyclone. That sounds like a good example.

Re: RNBO compatible external Max objects, I think there can be if you create an external entirely with RNBO compatible Max objects, though I haven't tried myself.

Also RNBO supports the patcher object, so IIUC in theory you should be able to use this to make an abstraction entirely of [RNBO compatible Max objects](https://rnbo.cycling74.com/objects), which can then be re-used in other RNBO patches and exported that way.

I'm not too familiar with the innards of the `nn~` external, but depending on how it was made and what Max objects it uses, perhaps it can be re-made with RNBO compatible Max objects for full RNBO export support.
